main.go

```diff
@@ -94,6 +95,11 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "CustomDomain")
 		os.Exit(1)
 	}
+
+	if err := managed.SetClusterVersion(mgr.GetClient()); err != nil {
+		setupLog.Error(err, "")
+		os.Exit(1)
+	}
```

Comment:
With the changes in `SetClusterVersion()` this should be removed - we should be doing this in the reconcile loop instead.